src/corppa/poetry_detection/annotation/recipe.py

```diff
@@ -234,12 +243,106 @@ def annotate_page_text(
         "config": config,
     }
 
+    # remove fetched image data before saving to the database
     if fetch_media:
-        components["before_db"] = lambda x: remove_images(x, image_prefix=image_prefix)
+        components["before_db"] = lambda x: remove_image_data(
+            x, image_prefix=image_prefix
+        )
 
     return components
 
 
+def get_session_name(example: TaskType, suffix: Optional[str] = None) -> str:
+    """
+    Extract session name from task examplke. Session ids have the following form:
+        [session id] = [db id]-[session name]
+    Assumes that session names do not contain dashes.
+    """
+    session_id = example[SESSION_ID_ATTR]
+    session_name = session_id.rsplit("-", maxsplit=1)[-1]
+    if suffix is not None:
+        session_name = f"{session_name}-{suffix}"
+    return session_name
+
+
+def add_label_prefix(label: str, prefix: str) -> str:
+    return f"{prefix}: {label}"
+
+
+def remove_label_prefix(label: str) -> str:
+    return label.rsplit(": ", maxsplit=1)[-1]
+
+
+def add_session_prefix(
+    example: TaskType, session_sfx: Optional[str] = None
+) -> TaskType:
+    """
+    Add session name as prefix to text span labels
+
+    label --> "[session]: label"
+    ex. "POETRY" --> "alice: POETRY"
+
+    Calls: `get_sesssion_name` and `add_label_prefix`
+    """
+    session_name = get_session_name(example, suffix=session_sfx)
+    for span in example.get("spans", []):
+        span["label"] = add_label_prefix(span["label"], session_name)
+    return example
+
+
+def remove_session_prefix(example: TaskType) -> TaskType:
+    """
+    Remove added session prefix from text span labels
+
+    Calls: `remove_label_prefix`
+    """
+    for span in example.get("spans", []):
+        span["label"] = remove_label_prefix(span["label"])
+    return example
+
+
+def has_span_overlap(example: TaskType, strip_label_pfx: bool = True) -> bool:
+    """
+    Check if example has overlapping (text) span annotations
+
+    Calls: `remove_label_prefix` (optionally)
+    """
+    if "spans" not in example:
+        return False
+    label_coverage = {}  # label (str) --> coverage (intspan)
+    for span in example["spans"]:
+        # Get label
+        label = span["label"]
+        if strip_label_pfx:
+            label = remove_label_prefix(label)
+        # Represent span's coverage as an intspan
+        # Note: intspans are closed intervals, while span character ranges are half-open
```

Comment:
what does "half-open" mean here?

Reply:
ah, I see! makes sense - that's why the minus one on the end

Reply:
Is the updated comment helpful?

Reply:
that notation isn't familiar enough to me to be that helpful here - can we just say that intspan includes the last character but span characters do not?

Reply:
fair enough, this is standard interval notation; but I can rephrase it. since these aren't actually characters but integer sets; i'll just say something about how intspan's range includes the ending index (as opposed to python indexing)

Reply:
How about now?

Reply:
I figured it was standard notation, but it's not something I use frequently. That's a helpful clarification, to differentiate from python indexing. Probably good to include both standard interval notation and your note about ending index.

Sorry to belabor this; it does seem important to be clear on since I expect we will be working with these spans in other contexts and I want to make sure I understand.

For my own understanding on your note about characters vs integer sets - it's integers that reference character based on indexes, right? I see now that I was sloppy in my language, was thinking of what the numbers represent rather than the numbers themselves

Reply:
your last revision is fine with me if you're happy with it!

Reply:
Thanks for the good conversation, I'll tweak the comment a bit more but it's definitely worth documenting.

Reply:
How's the added example look?

Reply:
this looks great! thank you